lib/fs.js

```diff
@@ -648,6 +654,14 @@ fs.readSync = function(fd, buffer, offset, length, position) {
     offset = 0;
   }
 
+  if (length === 0) {
```

Comment:
@trevnorris Ah, you are right. I totally missed that. Then this will not be a problem here.